raster/src/main/scala/geotrellis/raster/io/geotiff/GeoTiff.scala

```diff
@@ -52,6 +52,19 @@ trait GeoTiff[T <: CellGrid] extends GeoTiffData {
  * Companion object to GeoTiff
  */
 object GeoTiff {
+  def readMultiBand(path: String): MultibandGeoTiff =
+    MultibandGeoTiff(path)
+
+  def readSingleBand(path: String): SinglebandGeoTiff =
+    SinglebandGeoTiff(path)
+
+  def apply(path: String): Either[SinglebandGeoTiff, MultibandGeoTiff] =
+    if (MultibandGeoTiff(path).tile.bandCount == 1) {
+      Left(SinglebandGeoTiff(path))
+    } else {
+      Right(MultibandGeoTiff(path))
+    }
+
```

Comment:
We also should have overloads to build a GeoTiff from [`ProjectedTile`](https://github.com/geotrellis/geotrellis/blob/master/raster/src/main/scala/geotrellis/raster/ProjectedRaster.scala)